src/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs

```diff
@@ -45,6 +47,19 @@ protected KeyedCollection(IEqualityComparer<TKey> comparer, int dictionaryCreati
             _threshold = dictionaryCreationThreshold;
         }
 
+        /// <summary>
+        /// Enables the use of foreach internally without allocations using <see cref="List{T}"/>'s struct enumerator.
+        /// </summary>
+        new private List<TItem> Items
+        {
+            get
+            {
+                Debug.Assert(base.Items.GetType() == typeof(List<TItem>));
```

Comment:
List<T>.GetEnumerator() isn't virtual, so as called here, it couldn't. I'm just being pedantic in the opposite direction, though,